phylanx/execution_tree/primitives/base_primitive.hpp

```diff
@@ -177,7 +177,11 @@ namespace phylanx { namespace execution_tree { namespace primitives
         virtual primitive_result_type eval_direct(
             std::vector<primitive_argument_type> const& params) const
         {
-            return eval(params).get();
+            ++count_;
+            hpx::util::high_resolution_timer eval_timer(duration_);
+            const auto& ret = eval(params).get();
+            duration_ = eval_timer.elapsed_nanoseconds();
```

Comment:
This will collect the execution time once only, overwriting the previous timings (which have to be accumulated).

Reply:
`eval_time` is initialized as`eval_timer(duration_)`. Why is this a problem?